thorlcr/activities/indexread/thindexread.cpp

```diff
@@ -386,13 +403,17 @@ class CIndexCountActivityMaster : public CIndexReadBase
                 limit = (rowcount_t)helper->getChooseNLimit();
             else
                 totalCountKnown = true; // totalCount = 0
+            if (helper->canMatchAny())
+                limit = (rowcount_t)helper->getKeyedLimit();
```

Comment:
hm, I think I lost a change relating to this, will check.

Reply:
Fixed in new commit, I don't actually think it could cause problems.
I tried to create a test that it made a difference in but failed to.
It meant that choosen limit was being ignored in effect.
If the keyedlimit was higher, it would potentially send a higher total to the slave, but the slave would trim it to the choosen value anyway.
If keyedLimit was lower than choosen, then total couldn't exceed choosen anyway.